### DISCUSSION TITLE
order_by method incorrectly infers return type of select

### DISCUSSION THREAD

**Discussion by SergeyKapshuchenko** [fastapi/sqlmodel#854] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import Field, SQLModel, select


class Hero(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    age: int
    age_optional: int | None = Field(default=None)
    name: str


if __name__ == "__main__":
    query_1 = select(Hero).order_by(
        Hero.age
    )  # return type is Any if the field is required integer

    query_2 = select(Hero).order_by(
        Hero.age_optional
    )  # return type is SelectOfScalar[Hero] if the field is optional integer
```

### Description

When using a required integer field with order_by in SQLModel, the inferred return type is 'Any'

<img width="656" alt="Знімок екрана 2024-03-22 о 17 40 38" src="https://github.com/tiangolo/sqlmodel/assets/50737888/1ff01f48-83c8-46a9-a966-4ffcdd3ed9cf">
<img width="656" alt="Знімок екрана 2024-03-22 о 17 40 40" src="https://github.com/tiangolo/sqlmodel/assets/50737888/98cf2e08-68a9-491f-bfaa-1ca3f240ab09">

### Operating System

macOS

### SQLModel Version

0.0.16

### Python Version

3.12

**Comment by SergeyKapshuchenko** [comment]
any updates ?

**Comment by YuriiMotov** [comment]
Works fine in current version (0.0.24) - both types are `SelectOfScalar[Hero]`
